shared/js/search_provider.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+/* exported SearchProvider */
+/* globals LazyLoader */
+/* globals SettingsListener */
+
+(function(exports) {
+
+  // When the "search_providers.json" file is edited, this version
+  // should be bumped so that existing clients reload the updated
+  // data
+  var VERSION = 1;
+
+  // Cache for current provider configuration
+  var SEARCH_CACHE_KEY = 'search.cache';
+  // The users current provider selection
+  var SEARCH_PROVIDER_KEY = 'search.provider';
+
+  // File containing the provider configurations for all partners
+  var DEFAULT_PROVIDERS_URL = '/shared/js/search_providers.json';
```

Comment:
We will populate the settings value based on runtime information as specified in https://bugzilla.mozilla.org/show_bug.cgi?id=1125131